lib/authenticate_methods.rb

```diff
@@ -63,7 +63,7 @@ def authenticate_user!
   end
 
   def authenticate_admin!
-    return if current_user.is_administrator?
+    return if current_user.is_administrator? || !current_user.oauth_applications.empty?
```

Comment:
This is a pretty massive bug.  Makes any user with oauth applications an administrator across all of Accounts.

I know you added this because of our conversation where I said "check that they are an admin or have oauth applications" but we need to do that check in a _separate_ method that is only used for the specific controller we are modifying.  `authenticate_admin!` is used all over the place to authenticate that someone is an admin.

I think something like the following would do it.

```ruby
class ApplicationsController < Doorkeeper::ApplicationsController
  before_action :authenticate_admin_or_oauth_application_owner!

  ...

  protected

  def authenticate_admin_or_oauth_application_owner!
    return if current_user.oauth_applications.any?
    authenticate_admin! # calling this existing method lets us take advantage of
                        # how it sends users to the login page and back here if not logged in
  end
end
```

Reply:
I searched for admin_authentication which was the alias for authenticate_admin and it is only used in the controller I was editing. I neglected to search for authenticate_admin. I assumed since it was a doorkeeper method it was only a part of the Oauth Applications. I missed that is was used in BaseController. Thanks for catching this.